pkg/operator/apis/monitoring/v1alpha1/types_integrations.go

```diff
@@ -0,0 +1,122 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IntegrationsSubsystemSpec defines global settings to apply across the
+// integrations subsystem.
+type IntegrationsSubsystemSpec struct {
+	// Label selector to find Integration resources to run. When nil, no
+	// integration resources will be defined.
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+
+	// Label selector for namespaces to search when discovering integration
+	// resources. If nil, integration resources are only discovered in the
+	// namespace of the GrafanaAgent resource.
+	//
+	// Set to `{}` to search all namespaces.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path="integrations"
+// +kubebuilder:resource:singular="integration"
+// +kubebuilder:resource:categories="agent-operator"
+
+// Integration runs a single Grafana Agent integration. Integrations that
+// generate telemetry must be configured to send that telemetry somewhere; such
+// as autoscrape for exporter-based integrations.
+//
+// Integrations have access to the LogsInstances and MetricsInstances in the
+// same GrafanaAgent resource set, referenced by the <namespace>/<name> of the
+// *Instance resource.
+//
+// For example, if there is a default/production MetricsInstance, you can
+// configure a supported integration's autoscrape block with:
+//
+//     autoscrape:
+//       enable: true
+//       metrics_instance: default/production
+//
+// There is currently no way for telemetry created by an Operator-managed
+// integration to be collected from outside of the integration itself.
+type Integration struct {
+	metav1.TypeMeta   `json:",inline"`
```

Comment:
This works for bson, but does it work for json as well? 🤯

Reply:
After a discussion, it's probably for the best to be explicit about our need to inline TypeMeta here. Resolving this comment.